sentry_sdk/integrations/grpc/aio/client.py

```diff
@@ -20,20 +21,9 @@ class ClientInterceptor:
     def _update_client_call_details_metadata_from_scope(
         client_call_details: ClientCallDetails,
     ) -> ClientCallDetails:
-        metadata = (
-            list(client_call_details.metadata) if client_call_details.metadata else []
-        )
+        client_call_details.metadata = client_call_details.metadata or Metadata()
```

Comment:
Unfortunately, setting this attribute is not possible; see the following from our test logs:

```
tests/integrations/grpc/test_grpc_aio.py:76: in test_grpc_server_starts_transaction
    await stub.TestServe(gRPCTestMessage(text="test"))
.tox/py3.8-grpc-latest/lib/python3.8/site-packages/grpc/aio/_interceptor.py:470: in __await__
    call = yield from self._interceptors_task.__await__()
.tox/py3.8-grpc-latest/lib/python3.8/site-packages/grpc/aio/_interceptor.py:697: in _invoke
    return await _run_interceptor(
.tox/py3.8-grpc-latest/lib/python3.8/site-packages/grpc/aio/_interceptor.py:671: in _run_interceptor
    call_or_response = await interceptors[0].intercept_unary_unary(
sentry_sdk/integrations/grpc/aio/client.py:47: in intercept_unary_unary
    client_call_details = self._update_client_call_details_metadata_from_scope(
sentry_sdk/integrations/grpc/aio/client.py:24: in _update_client_call_details_metadata_from_scope
    client_call_details.metadata = client_call_details.metadata or Metadata()
E   AttributeError: can't set attribute
```

Perhaps, `ClientCallDetails` provides some other way to set this?

Reply:
Thank you for pointing this out. I remembered there was a problem with setting some attribute and thought it was in `add` itself.

`ClienCallDetails` is a subclass of `collections.namedtuple`, which is immutable so it needs to be reconstructed to change a field value. This can be done with `._replace`. Was hesitating to use it before as it is prefixed with `_`, however, according to [the docs](https://docs.python.org/3/library/collections.html#namedtuple-factory-function-for-tuples-with-named-fields), this is just to avoid name clashes with fields.

Pushed a fixup, please have a look.

Reply:
Tests failing due to something else, will have a look.

Reply:
The metadata was not guaranteed to be an `Optional[Metadata]` object, as it is typed, up until grpcio 1.65.0 and a tuple was mistakenly passed on through the interceptors. This was fixed on the gRPC side.

Do you want to keep support for versions before that?

Reply:
Yes, we should keep support for anything we currently support, as this would be a breaking change that requires a major version bump if we drop support for any gRPC versions.

Reply:
Added the check as it is part of grpcio itself for newer versions :+1:

Reply:
[This](https://github.com/grpc/grpc/pull/34347/commits/8f50070251c530acaee237d1f3841cfc61df82a9) is  the relevant change in the grpc package